src/models/BrowserStyleSheet.js

```diff
@@ -0,0 +1,123 @@
+// @flow
+/*
+ * Browser Style Sheet with Rehydration
+ *
+ * <style data-styled-components="x y z"
+ *        data-styled-components-is-local="true">
+ *   /· sc-component-id: a ·/
+ *   .sc-a { ... }
+ *   .x { ... }
+ *   /· sc-component-id: b ·/
+ *   .sc-b { ... }
+ *   .y { ... }
+ *   .z { ... }
+ * </style>
+ *
+ * Note: replace · with * in the above snippet.
+ * */
+import extractCompsFromCSS from '../utils/extractCompsFromCSS'
+import type { Tag } from './StyleSheet'
+import StyleSheet, { SC_ATTR, LOCAL_ATTR } from './StyleSheet'
+
+export const COMPONENTS_PER_TAG = 40
+
+class BrowserTag implements Tag {
+  isLocal: boolean
+  components: Map<string, Object>
+  el: HTMLElement
+  ready: boolean
+
+  constructor(el: HTMLElement, isLocal: boolean, existingSource: string = '') {
+    this.el = el
+    this.isLocal = isLocal
+    this.components = new Map(extractCompsFromCSS(existingSource).map(obj =>
```

Comment:
Actually, found this thing here. `Map` is not polyfilled by us, so in environments where the user is not polyfilling it, it might cause issues for a wide variety of old browsers.